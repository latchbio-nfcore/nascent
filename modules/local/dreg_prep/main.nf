process DREG_PREP {

    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-f01e242bdea19948f0576fdca94777242fe4c2cb:4238fb992d2a93e648108c86e3a9f51348e834a9-0' :
        'biocontainers/mulled-v2-f01e242bdea19948f0576fdca94777242fe4c2cb:4238fb992d2a93e648108c86e3a9f51348e834a9-0' }"

    input:
    tuple val(meta), path(bam_file), val(index)
    path  sizes
    val assay_type

    output:
    tuple val(meta), path("${prefix}.pos.bw"), path("${prefix}.neg.bw"), emit: dreg_bigwig
    tuple val(meta), path("${prefix}.bedGraph"), emit: dreg_bg

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    if (meta.single_end) {
        template "proseq2.0_se"
    } else {
        template "proseq2.0_pe"
    }
}
