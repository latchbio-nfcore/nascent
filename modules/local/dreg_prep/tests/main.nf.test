nextflow_process {
    name "Test Process DREG_PREP"
    script "../main.nf"
    process "DREG_PREP"
    // config "./nextflow.config"

    tag "modules"
    tag "dreg"

    test("sarscov2 - paired-end") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end: false ], // meta map
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ])
                // sizes
                input[1] = file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.sizes', checkIfExists: true)
                // extension
                input[2] = 'R1_5'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match("paired-end") }
            )
        }

    }

    test("sarscov2 - single-end") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end: true ], // meta map
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/bam/test.single_end.sorted.bam', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/bam/test.single_end.sorted.bam.bai', checkIfExists: true)
                ])
                // sizes
                input[1] = file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.sizes', checkIfExists: true)
                // extension
                input[2] = 'GROseq'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match("single-end") }
            )
        }

    }
}
