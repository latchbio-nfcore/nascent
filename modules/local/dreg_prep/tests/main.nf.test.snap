{
    "single-end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.dreg_plus.rpm.bw:md5,c723fc5bdb582655ea1aa1f324cb71b4"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.dreg_minus.rpm.bw:md5,2c78d94d8abc4a2bad792fd597d434be"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.dreg_plus.bw:md5,fd77f4d72b9f77ed71166d21de1b48bd"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.dreg_minus.bw:md5,92c7a60a606a1668a388c3f45d8f07f0"
                    ]
                ],
                "minus_bigwig": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.dreg_minus.bw:md5,92c7a60a606a1668a388c3f45d8f07f0"
                    ]
                ],
                "minus_rpm_bigwig": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.dreg_minus.rpm.bw:md5,2c78d94d8abc4a2bad792fd597d434be"
                    ]
                ],
                "plus_bigwig": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.dreg_plus.bw:md5,fd77f4d72b9f77ed71166d21de1b48bd"
                    ]
                ],
                "plus_rpm_bigwig": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.dreg_plus.rpm.bw:md5,c723fc5bdb582655ea1aa1f324cb71b4"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-30T14:19:05.327654675"
    },
    "paired-end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dreg_plus.rpm.bw:md5,289652b9b07dec1d0c402e0de9001d9b"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dreg_minus.rpm.bw:md5,5c7c70cecb2ff37236dffa8cd6e64e8f"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dreg_plus.bw:md5,7098afd4d0a3927a97bca0f390318ef0"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dreg_minus.bw:md5,6fc8a8d47c17b9c3e9fa00302cf4b388"
                    ]
                ],
                "minus_bigwig": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dreg_minus.bw:md5,6fc8a8d47c17b9c3e9fa00302cf4b388"
                    ]
                ],
                "minus_rpm_bigwig": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dreg_minus.rpm.bw:md5,5c7c70cecb2ff37236dffa8cd6e64e8f"
                    ]
                ],
                "plus_bigwig": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dreg_plus.bw:md5,7098afd4d0a3927a97bca0f390318ef0"
                    ]
                ],
                "plus_rpm_bigwig": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.dreg_plus.rpm.bw:md5,289652b9b07dec1d0c402e0de9001d9b"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-30T14:18:55.16822952"
    }
}